sklearn/ensemble/_hist_gradient_boosting/grower.py

```diff
@@ -546,26 +569,32 @@ def _fill_predictor_node_array(predictor_nodes, grower_node,
         node['feature_idx'] = feature_idx
         node['bin_threshold'] = bin_idx
```

Comment:
this thing becomes incorrect when we have a categorical split.

Also, we should try to keep the symmetry between numerical and categorical split: when `bin_tresholds=None` and the split is numerical,  the `bin_threshold` field is still set (on this line). So `predict_binned()` still works. But when the split is categorical, the categorical bitset won't be set and `predict_binned()` won't work.

Reply:
> this thing becomes incorrect when we have a categorical split.

It doesn't have a meaning when the feature is categorical. Are you thinking of not setting `bin_threshold` at all?

> Also, we should try to keep the symmetry between numerical and categorical split:

I have been considering `bin_thresholds=None` used purely for testing. In other words, it is only set to `None` in `test_predictor_from_grower` and `test_missing_value_predict_only`. In this case, without `bin_thresholds`, `predict_binned` shouldn't be able to work with categories. Since we always set `bin_thresholds` in non-testing code, I do not think we need to think of a good default with `bin_thresholds=None`.

Reply:
> In this case, without bin_thresholds, predict_binned shouldn't be able to work with categories

I would expect it to work since it should be able to leverage the bitset that's in the node's `split_info`? I'm writing a unit test for the grower comparing with OHE (which seems to pass \o/), and this was surprising to me.

I guess it's OK not to support it if it makes things too complex, but in this case I think we should explicitly raise an error saying "you can't have categorical splits and not pass the thresholds: neither predict() or predict_binned() will work."

Also semi-related: please take a look at https://github.com/scikit-learn/scikit-learn/pull/18179 ;)

Reply:
> In this case, without bin_thresholds, predict_binned shouldn't be able to work with categories

Whoops, I meant `predict` would not work.

We can kind of lower the complexity, by setting `bin_thresholds` in tests as well.